share/empty.go

```diff
@@ -3,24 +3,33 @@ package share
 import (
 	"bytes"
 	"context"
+	"fmt"
+	"math"
 
 	"github.com/ipfs/go-blockservice"
 
 	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/pkg/da"
+	"github.com/celestiaorg/rsmt2d"
 )
 
 // EnsureEmptySquareExists checks if the given DAG contains an empty block data square.
 // If it does not, it stores an empty block. This optimization exists to prevent
 // redundant storing of empty block data so that it is only stored once and returned
 // upon request for a block with an empty data square. Ref: header/constructors.go#L56
-func EnsureEmptySquareExists(ctx context.Context, bServ blockservice.BlockService) error {
-	shares := make([][]byte, appconsts.MinShareCount)
-	for i := 0; i < appconsts.MinShareCount; i++ {
-		shares[i] = tailPaddingShare
-	}
+func EnsureEmptySquareExists(ctx context.Context, bServ blockservice.BlockService) (*rsmt2d.ExtendedDataSquare, error) {
+	return AddShares(ctx, emptyDataSquare(), bServ)
+}
 
-	_, err := AddShares(ctx, shares, bServ)
-	return err
+// EmptyExtendedDataSquare returns the EDS of the empty block data square.
+func EmptyExtendedDataSquare() *rsmt2d.ExtendedDataSquare {
+	shares := emptyDataSquare()
+	squareSize := uint64(math.Sqrt(float64(appconsts.MinSquareSize)))
+	eds, err := da.ExtendShares(squareSize, shares)
+	if err != nil {
+		panic(fmt.Errorf("failed to create empty EDS: %w", err))
+	}
```

Comment:
why panic and not just return an error here?

Reply:
Only ever gets called on startup, and if we cant create an empty EDS something is VERY wrong

Reply:
But if we return an error, then the node will not start. Or it is mostly done to be able to identify this case?